File: api_nestjs_distributed_order_platform/src/application/use-case/stock/get-stock.use-case.ts
import { Inject, Injectable, NotFoundException } from "@nestjs/common";
import type { StockRepositoryInterface } from "src/application/port/stock.repository";
import { StockEntity } from "src/domain/entities/stock.entity";
import { STOCK_REPOSITORY } from "src/shared/tokens_nest/stock.token";

@Injectable()
export class GetStockUseCase {
    constructor(
        @Inject(STOCK_REPOSITORY)
        private readonly stockRepository: StockRepositoryInterface,
    ) {}

    async execute(productId: string): Promise<StockEntity> {
        const stock = await this.stockRepository.findByProductId(productId);
        if (!stock) {
            throw new NotFoundException('Stock not found');
        }
        return stock;
    }
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/usecase/stock/GetStockUseCase.java
package com.project.order.application.usecase.stock;

import com.project.order.application.exception.NotFoundException;
import com.project.order.application.port.StockRepository;
import com.project.order.domain.model.Stock;
import org.springframework.stereotype.Service;

@Service
public class GetStockUseCase {

    private final StockRepository stockRepository;

    public GetStockUseCase(StockRepository stockRepository) {
        this.stockRepository = stockRepository;
    }

    public Stock execute(String productId) {
        return stockRepository.findById(productId)
                .orElseThrow(() -> new NotFoundException("Stock not found"));
    }
}
